tests/e2e/system_test/test_mcg_replication_with_disruptions.py

```diff
@@ -0,0 +1,248 @@
+        compare_bucket_object_list(
+            mcg_obj_session, source_bucket_name, target_bucket_name
+        )
+
+        logger.info("Bi directional bucket replication working as expected")
+
+        # write some object to any of the bucket, followed by immediate cluster restart
+        written_random_objects = write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            target_bucket_name,
+            test_directory_setup.origin_dir,
+            mcg_obj=mcg_obj_session,
+            amount=1,
+            pattern="third-object-",
+        )
+        print(f"Written objects: {written_random_objects}")
+
+        node_list = get_worker_nodes()
+        node_objs = get_node_objs(node_list)
+        nodes.restart_nodes(node_objs, force=False, timeout=500)
+
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15,
+        )(ocp.wait_for_cluster_connectivity(tries=400))
+
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15,
+        )(wait_for_nodes_status(timeout=1800))
+
+        wait_for_pods_to_be_running(namespace=config.ENV_DATA["cluster_namespace"])
+
+        logger.info("Nodes rebooted successfully!!")
+
+        compare_bucket_object_list(
+            mcg_obj_session, source_bucket_name, target_bucket_name
+        )
+
+        logger.info("Objects sync works even when the cluster is rebooted")
+
+        # delete all the s3-compatible namespace buckets objects and then recover it from aws namespace bucket on write
+        rgw_objects = [
+            obj.key
+            for obj in mcg_obj_session.s3_list_all_objects_in_bucket(target_bucket_name)
+        ]
+        logger.info(f"RGW objects: {rgw_objects}")
+
+        for obj in rgw_objects:
+            s3_delete_object(
+                mcg_obj_session, bucketname=target_bucket_name, object_key=obj
+            )
+
+        obj_after_deletion = [
+            obj.key
+            for obj in mcg_obj_session.s3_list_all_objects_in_bucket(target_bucket_name)
+        ]
+        logger.info(f"Objects after deletion: {obj_after_deletion}")
+        logger.info("All the objects in RGW namespace buckets are deleted!!!")
+
+        written_random_objects = write_random_test_objects_to_bucket(
+            awscli_pod_session,
+            target_bucket_name,
+            test_directory_setup.origin_dir,
+            mcg_obj=mcg_obj_session,
+            amount=1,
+            pattern="fourth-object-",
+        )
+        print(f"Written objects: {written_random_objects}")
+
+        compare_bucket_object_list(
+            mcg_obj_session, source_bucket_name, target_bucket_name
+        )
+
+        logger.info(
+            "All the objects retrieved back to s3-compatible bucket on new write to aws-s3 bucket!!"
```

Comment:
From what I understand, this part deletes all of the objects in one bucket, thus removing them all in the second one too.
Then, it writes new objects and verifies they were replicated. Is this the intended flow?

Reply:
> From what I understand, this part deletes all of the objects in one bucket, thus removing them all in the second one too.
deleting objects from one bucket wont delete them on second bucket, afaik.

Reply:
@MeridianExplorer